allennlp/training/trainer.py

```diff
@@ -493,7 +778,18 @@ def _restore_checkpoint(self) -> Tuple[int, List[float]]:
         else:
             val_metric_per_epoch = training_state["val_metric_per_epoch"]
 
-        return training_state["epoch"] + 1, val_metric_per_epoch
+        if isinstance(training_state["epoch"], int):
+            epoch_to_return = training_state["epoch"] + 1
+        else:
+            epoch_to_return = int(training_state["epoch"].split('.')[0]) + 1
+
+        # For older checkpoints with batch_num_total missing, default to old behavior where
+        # it is unchanged.
+        batch_num_total = training_state.get('batch_num_total')
+        if batch_num_total is not None:
+            self._batch_num_total = batch_num_total
```

Comment:
Oh I kind of see why you did this now. It might be better to just pass this value to `train_epoch` though, rather than setting it as state. This way you could load a model, get the return value from this function, set it in trainer, and then pass it to `train_epoch`.